src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -242,6 +262,35 @@ module Make (Inputs : Inputs_intf) :
     List.bind (successors t breadcrumb) ~f:(fun succ ->
         succ :: successors_rec t succ )
 
+  let compute_new_successor_root_hash t parent_node =
+    let new_length = parent_node.length + 1 in
+    let root_node = Hashtbl.find_exn t.table t.root in
+    let root_hash = With_hash.hash root_node.breadcrumb.transition_with_hash in
+    let distance_to_root = new_length - root_node.length in
+    if distance_to_root > max_length then
+      `Changed_root (List.hd_exn (path t parent_node.breadcrumb))
+    else `Same_root root_hash
+
+  let create_external_transition t ~protocol_state ~protocol_state_proof
```

Comment:
I think having this portion of code live in transition frontier is misleading as this operation is only performed from the proposer. The transition frontier should just provide a hook to determine what the new root hash would be if an external transition were to be added to the tree.